### PR TITLE
nad pkg added IpVlanPlugin and whereabout IPAM

### PR DESCRIPTION
**Pull request by kononovn**

### CODE REVIEW

File: pkg/nad/ipams.go [hunk 1]
```diff
@@ -4,3 +4,33 @@ package nad
 func IPAMStatic() *IPAM {
 	return &IPAM{Type: "static"}
 }
+
+// IPAMWhereAbout returns WhereAbout ipam type.
```
Comment (evgenLevin) [inline pkg/nad/ipams.go:8]: Whereabouts

File: pkg/nad/ipams.go [hunk 2]
```diff
@@ -4,3 +4,33 @@ package nad
...
+func IPAMWhereAbout(ipRange, gateway string) *IPAM {
+	ipam := &IPAM{Type: "whereabouts", IPRanges: []IPRanges{{Range: ipRange, Gateway: gateway}}}
+
+	if ipRange == "" {
+		return nil
+	}
+
+	if gateway == "" {
```
Comment (evgenLevin) [inline pkg/nad/ipams.go:16]: better to run all verfication and only after that create IPAM:
```
if ipRange == "" {
		return nil
	}

	if gateway == "" {
			return nil
	}
	
return &IPAM{Type: "whereabouts", IPRanges: []IPRanges{{Range: ipRange, Gateway: gateway}}}
```

File: pkg/nad/ipams.go [hunk 3]
```diff
@@ -4,3 +4,33 @@ package nad
...
+func WhereAboutAppendRange(ipam *IPAM, ipRange, gateway string) *IPAM {
+	ipam.IPRanges = append(ipam.IPRanges, IPRanges{Range: ipRange, Gateway: gateway})
+
+	if ipRange == "" {
+		return nil
+	}
+
+	if gateway == "" {
+		return nil
+	}
```
Comment (evgenLevin) [inline pkg/nad/ipams.go:33]: also here, please verify that is no nill and after that modify ipam object

File: pkg/nad/masterplugin.go [hunk 4]
```diff
@@ -290,3 +290,106 @@ func (plugin *MasterVlanPlugin) GetMasterPluginConfig() (*MasterPlugin, error) {
...
 	return plugin.masterPlugin, nil
 }
+
+// MasterIPVlanPlugin provides struct for MasterPlugin set to vlan in NetworkAttachmentDefinition.
+type MasterIPVlanPlugin struct {
+	masterPlugin *MasterPlugin
+	errorMsg     string
+}
+
+// NewMasterIPVlanPlugin creates new instance of MasterVlanPlugin.
```
Comment (evgenLevin) [inline pkg/nad/masterplugin.go:300]: MasterIPVlanPlugin.

File: pkg/nad/masterplugin.go [hunk 5]
```diff
@@ -290,3 +290,106 @@ func (plugin *MasterVlanPlugin) GetMasterPluginConfig() (*MasterPlugin, error) {
 
 	return plugin.masterPlugin, nil
 }
+
+// MasterIPVlanPlugin provides struct for MasterPlugin set to vlan in NetworkAttachmentDefinition.
```
Comment (evgenLevin) [inline pkg/nad/masterplugin.go:294]: .. set to IP vlan ... ?

File: pkg/nad/masterplugin.go [hunk 6]
```diff
@@ -290,3 +290,106 @@ func (plugin *MasterVlanPlugin) GetMasterPluginConfig() (*MasterPlugin, error) {
...
+type MasterIPVlanPlugin struct {
+	masterPlugin *MasterPlugin
+	errorMsg     string
+}
+
+// NewMasterIPVlanPlugin creates new instance of MasterIP	VlanPlugin.
```
Comment (evgenLevin) [inline pkg/nad/masterplugin.go:300]: remove space

File: pkg/nad/masterplugin.go [hunk 7]
```diff
@@ -290,3 +290,106 @@ func (plugin *MasterVlanPlugin) GetMasterPluginConfig() (*MasterPlugin, error) {
...
+func NewMasterIPVlanPlugin(name string) *MasterIPVlanPlugin {
+	glog.V(100).Infof(
+		"Initializing new MasterVlanPlugin structure %s", name)
```
Comment (evgenLevin) [inline pkg/nad/masterplugin.go:303]: MasterIPVlanPlugin

**Comment by kononovn** [comment]
@evgenLevin PTAL

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**